pkg/hive/manager.go

```diff
@@ -51,6 +52,31 @@ type clusterManager struct {
 	dh dynamichelper.Interface
 }
 
+// NewFromEnv can return a nil ClusterManager when hive features are disabled. This exists to support regions where we don't have hive,
+// and we do not want to restrict the frontend from starting up successfully.
+// It has the caveat of requiring a nil check on any operations performed with the returned ClusterManager
+// until this conditional return is removed (we have hive everywhere).
+func NewFromEnv(ctx context.Context, log *logrus.Entry, env env.Interface) (ClusterManager, error) {
+	adoptByHive, err := env.LiveConfig().AdoptByHive(ctx)
+	if err != nil {
+		return nil, err
+	}
+	installViaHive, err := env.LiveConfig().InstallViaHive(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if !adoptByHive && !installViaHive {
```

Comment:
Should this be `!adoptByHive` && `!installViaHive`?  Shouldn't it be one or the other?

Reply:
Nevermind this logic is good.  It's hard to read sometimes :)